### PR TITLE
Pillows support **kwargs

### PR DESCRIPTION
**Pull request by emord**
It's too hard to follow the changes in #15773 so going to be splitting it into multiple PRs. and doing a little refactoring from it

This enables passing the number of processes and process number to a pillow. For example if the kafka-ucr-main pillow has a number of processes of 4, then we would expect the following to be running:

```
./manage.py run_ptop --pillow-name kafka-ucr-main --num-processes 4 --process-num 0
./manage.py run_ptop --pillow-name kafka-ucr-main --num-processes 4 --process-num 1
./manage.py run_ptop --pillow-name kafka-ucr-main --num-processes 4 --process-num 2
./manage.py run_ptop --pillow-name kafka-ucr-main --num-processes 4 --process-num 3
```

For now everything will only be running one process, but this will enable merging my multiple pillow machine deploy PRs https://github.com/dimagi/commcare-hq/pull/15854 https://github.com/dimagi/commcare-hq-deploy/pull/242

@dimagi/scale-team buddy @dannyroberts

### CODE REVIEW

**Comment by dannyroberts** [comment]
Just trying to wrap my head around the vision of the rollout process. If you wanted to merge this as a conceptually no-op change that just made rolling out the config change not crash but also not doing anything—wouldn't merging https://github.com/dimagi/commcare-hq-deploy/pull/242 end up initially starting 4 pillow processes each of which traverses the entirety of the (head of) the changes feed, when the desired effect is to initially still only start one process and down the road flip it to start 4 each of which only processes 1/4 of the changes?

**Comment by emord** [comment]
This PR is intended as a no-op change. The rollout process will look roughly like:

1. roll this out, which will allow to split pillows between machines and it won't crash if extra arguments are passed in.
2. start using `KafkaCheckpoint` as the source of truth for checkpoints. (it's current `DjangoPillowCheckpoint`, but it's writing duplicate data to KafkaCheckpoint)
3. implement handling multiple processes. (started in the other PR, but it's hard to follow and tests were failing , so I want to split it up to enable better code review and easier to reason about changes)
4. change currently split pillows to use num_processes instead of using the split ucr pillows (the main ucr pillows on prod, all ucr pillows on ICDS)
5. split up more of the pillows that fall behind at peak loads (should just be on icds)

I'm not sure where you're getting the 4 from? Each of those pillows have num_processes defined as 1, so for each pillow it should run:

`/manage.py run_ptop --pillow-name <pillow_name> --num-processes 1 --process-num 0`

**Review by snopoke — APPROVED** [review]

**Comment by dannyroberts** [comment]
Ah okay, that makes sense. I thought you were going to make settings that would run

```
./manage.py run_ptop --pillow-name kafka-ucr-main --num-processes 4 --process-num 0
./manage.py run_ptop --pillow-name kafka-ucr-main --num-processes 4 --process-num 1
./manage.py run_ptop --pillow-name kafka-ucr-main --num-processes 4 --process-num 2
./manage.py run_ptop --pillow-name kafka-ucr-main --num-processes 4 --process-num 3
```

but if you did that before those params really did anything, then there would be for processes each traversing all changes. With a single process with `--num-processes 1 --process-num 0` then there's not that problem.

Thanks!

**Review by dannyroberts — APPROVED** [review]
